### ISSUE TITLE
State usage

### ISSUE THREAD

**Issue by rafapaezbas** [fuzz-lightyear/fuzz-lightyear#7]
To share the state between operations and validations, should it be passed in the setup function https://github.com/fuzz-lightyear/fuzz-lightyear/blob/master/index.js#L80 ?

Something like: 

``` js
const operations = this._userFunctions.operations(reference, actual, this.rng, state, this.opts)
const validation = this._userFunctions.validation(reference, actual, this.rng, state, this.opts)
```

That would change the api of the `operations` and `validations` functions.

It could be done like:

``` js
const operations = this._userFunctions.operations(reference, actual, this.rng, this.opts, state)
const validation = this._userFunctions.validation(reference, actual, this.rng, this.opts, state)
```

in order to maintain compatibility.

**Comment by andrewosh** [comment]
@rafapaezbas yeah I think `state` isn't being tracked at all right now. Let me update the interface for you -- there's no reason for fuzz-lightyear to be opinionated about the `(reference, actual, state)` pattern. `setup` should just return some value that gets passed to `operations` and `validations`, then if you want to use the reference/actual pattern you can do that yourself.

**Comment by andrewosh** [comment]
This is fixed in v2.0.0 now, which is much less opinionated about the fuzzing state you use. Instead, your fuzzing module just returns operations + validators that can capture whatever state you want in a closure.
